src/Microsoft.Extensions.Caching.Abstractions/DistributedCacheExtensions.cs

```diff
@@ -34,14 +40,115 @@ public static void Set(this IDistributedCache cache, string key, byte[] value)
         /// <param name="key">The key to store the data in.</param>
         /// <param name="value">The data to store in the cache.</param>
         /// <returns>A task that represents the asynchronous set operation.</returns>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="key"/> or <paramref name="value"/> is null.</exception>
         public static Task SetAsync(this IDistributedCache cache, string key, byte[] value)
         {
             if (key == null)
             {
                 throw new ArgumentNullException(nameof(key));
             }
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
 
             return cache.SetAsync(key, value, new DistributedCacheEntryOptions());
         }
+
+        /// <summary>
+        /// Sets a string in the specified cache with the specified key.
+        /// </summary>
+        /// <param name="cache">The cache in which to store the data.</param>
+        /// <param name="key">The key to store the data in.</param>
+        /// <param name="value">The data to store in the cache.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="key"/> or <paramref name="value"/> is null.</exception>
+        public static void SetString(this IDistributedCache cache, string key, string value)
+        {
+            if (value == null)
```

Comment:
It's strange to just check the value here. I assume you depend on the called function to check the key? If so, you can do the same for value.